src/helpers/Location.ts

```diff
@@ -8,35 +8,37 @@
  *
  * class Location extends URL {
  *   get url() {
- *     return this.pathname + this.search;
+ *	 return this.pathname + this.search;
  *   }
  * }
  *
  */
 
-export default class Location {
-	constructor(url, base = document.baseURI) {
-		const location = new URL(url.toString(), base);
-		Object.defineProperty(location, 'url', { get: () => location.pathname + location.search });
-		return location;
+export default class Location extends URL {
+	constructor(url: string, base: string = document.baseURI) {
+		super(url.toString(), base);
+	}
+
+	get url() {
+		return this.pathname + this.search;
 	}
 
 	/**
 	 * Instantiate a Location from an element's href attribute
 	 * @param {Element} el
 	 * @return new Location instance
 	 */
-	static fromElement(el) {
+	static fromElement(el: HTMLAnchorElement): Location {
 		const href = el.getAttribute('href') || el.getAttribute('xlink:href');
-		return new Location(href);
+		return new Location(href!);
```

Comment:
Since we're getting the value with `getAttribute`, it might be `null`. `!` basically tells TS that we know it will be of some value (in this case `string`). 
This is not the best and it would be better to throw error if something like that happens, because it shouldn't, but that would be changing the implementation, so I avoided that.

Reply:
Sounds good!